packages/compiler/test/checker/intersections.test.ts

```diff
@@ -48,15 +49,20 @@ describe("compiler: intersections", () => {
   });
 
   it("emit diagnostic if one of the intersected type is not a model", async () => {
-    const diagnostics = await runner.diagnose(`
-      @test model Foo {
-        prop: {a: string} & "string literal";
-      }
-    `);
+    let source = `@test model Foo {
+      prop: {a: string} & ┆"string literal"┆;
+    }`;
+
+    let pos: number, end: number;
+    ({ source, pos } = extractCursor(source));
```

Comment:
feel like we might want more test like that, should we have another helper method that extract both end and start cursor?

Reply:
We should. If I hit it again, I'll do it unless you beat me to it :)